scripts/fast-reboot

```diff
@@ -117,7 +121,7 @@ function wait_for_pre_shutdown_complete_or_fail()
     waitcount=0
     # Wait up to 60 seconds for pre-shutdown to complete
     while [[ ${waitcount} -lt 600 ]]; do
-        STATE=`/usr/bin/redis-cli -n 6 hget "${TABLE}" state`
+        STATE=`timeout 10 /usr/bin/redis-cli -n 6 hget "${TABLE}" state`
```

Comment:
If timeout happened,  STATE might be in a unknown state? Might be better to check the timeout exist code before assigning the value.

Reply:
STATE won't be unknown per se. It would previously be empty before 3rd commit. but that is hard to tell from if the database key was never set.

Reply:
right, my point was that we should check the shell command (with timeout) exit code before we assign the value.  I saw you already add such logic in later commit.

Reply:
You are absolutely right :-)